src/Storages/StorageMerge.cpp

```diff
@@ -341,7 +339,7 @@ Pipe StorageMerge::createSources(
     if (!pipe.empty())
     {
         if (concat_streams && pipe.numOutputPorts() > 1)
-            pipe.addTransform(std::make_shared<ConcatProcessor>(pipe.getHeader(), pipe.numOutputPorts()));
```

Comment:
For this case, `ConcatProcessor` may be better then `Resize`, because we may have many tables we read from `merge`, and it may cause, e.g. too many files opened at the same time.

Reply:
Ah, that's reasonable.